store/persist.ts

```diff
@@ -17,8 +17,51 @@ export interface PersistProps<S extends AnyState> {
   key: string;
   reconciler: (original: S, rehydrated: Partial<S>) => S;
   rehydrate: () => Operation<Result<unknown>>;
+  transform?: TransformFunctions<S>;
 }
 
+interface TransformFunctions<S extends AnyState> {
+  in(s: Partial<S>): Operation<Partial<S>>;
+  out(s: Partial<S>): Operation<Partial<S>>;
+}
+
+export function createTransform<S extends AnyState>() {
+  const transformers: TransformFunctions<S> = {
+    in: function* (currentState: Partial<S>): Operation<Partial<S>> {
+      return currentState;
+    },
+    out: function* (currentState: Partial<S>): Operation<Partial<S>> {
+      return currentState;
+    },
+  };
+  const setInTransformer = function (
+    transformer: (fn: Partial<S>) => Operation<Partial<S>>,
+  ): void {
+    transformers.in = transformer;
+  };
+  const setOutTransformer = function (
+    transformer: (fn: Partial<S>) => Operation<Partial<S>>,
+  ): void {
+    transformers.out = transformer;
+  };
```

Comment:
I thought about the possibility of defining the transformer function depending on a dynamic element. For example "userRights" - certain tables are transformed depending on access rights.
It can also be seen as over-engineering, without contesting. But I considered that it gives a high degree of freedom.

Reply:
I like how you are thinking ahead, so don't change that mentality.  However, I think the API would be simpler if we just made a one-time set when creating the transformer.

Reply:
I've simplified the API by removing the dynamic setters (`setInTransformer` and `setOutTransformer`) to address the feedback about API complexity. This version still maintains flexibility by allowing state transformations, without the need for dynamic changes after the initial creation.
If you think that further simplifications are necessary, I would be happy to refactor.